### PR TITLE
Merge sync pointer restarts

### PR DESCRIPTION
**Pull request by avalonche**
Fixes | Closes | Resolves #

> Anything marked as optional that you didn't need to fill in, please remove it from the PR description. Choose one of the keywords above to refer to the issue this PR solves, followed by the issue number (e.g Fixes # 666). If there is no issue, remove the line. Remove this note after reading.

## Changes:
-

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe): 

## Testing
**Requires testing**

- [x] Yes
- [ ] No

**In case you checked yes, did you write tests??**

- [ ] Yes
- [x] No

**Comments about testing , should you have some** (optional)

Need to fix tests

## Further comments (optional)

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: src/Nethermind/Nethermind.Db/MetadataDbKeys.cs [hunk 1]
```diff
@@ -22,11 +22,9 @@ public static class MetadataDbKeys
         public const int TerminalPoWHash = 1;
         public const int TerminalPoWNumber = 2;
         public const int FinalizedBlockHash = 3;
-        public const int SafeBlockHash = 4;
```
Comment (MarekM25) [inline src/Nethermind/Nethermind.Db/MetadataDbKeys.cs:25]: Why we remove SafeBlockHash?

File: src/Nethermind/Nethermind.Merge.Plugin/Synchronization/BeaconHeadersSyncFeed.cs [hunk 2]
```diff
@@ -82,21 +88,25 @@ public override void InitializeFeed()
 
     protected override void FinishAndCleanUp()
     {
-        // TODO: beaconsync backfill of TD should be moved to forward beacon sync
-        if (_mergeConfig.FinalTotalDifficultyParsed == null)
-        {
-            // set total difficulty as beacon pivot does not provide total difficulty
```
Comment (MarekM25) [inline src/Nethermind/Nethermind.Merge.Plugin/Synchronization/BeaconHeadersSyncFeed.cs:88]: we can remove this comment now right?

File: src/Nethermind/Nethermind.Merge.Plugin/Synchronization/BeaconHeadersSyncFeed.cs [hunk 3]
```diff
@@ -111,22 +121,22 @@ protected override AddBlockResult InsertToBlockTree(BlockHeader header)
             // if (_blockTree.LowestInsertedHeader != null
             //     && _blockTree.LowestInsertedHeader.Number < (_blockTree.LowestInsertedBeaconHeader?.Number ?? long.MaxValue))
```
Comment (MarekM25) [inline src/Nethermind/Nethermind.Merge.Plugin/Synchronization/BeaconHeadersSyncFeed.cs:122]: hmmm do you think we can revert this if comment?

Comment (avalonche) [reply src/Nethermind/Nethermind.Merge.Plugin/Synchronization/BeaconHeadersSyncFeed.cs:122]: we can revert in another branch

**Comment by avalonche** [comment]
merged as part of `forkchoiceUpdate_changes` branch

**Pull request closed without merging**